teammapper-backend/src/map/services/maps.service.ts

```diff
@@ -140,23 +137,22 @@ export class MapsService {
   async updateNode(
     mapId: string,
     clientNode: IMmpClientNode
-  ): Promise<MmpNode> {
-    const existingNode = await this.nodesRepository.findOne({
-      where: { nodeMapId: mapId, id: clientNode.id },
-    })
-
-    if (!existingNode) return Promise.reject()
+  ): Promise<MmpNode | null> {
+    const mmpNode = mapClientNodeToMmpNode(clientNode, mapId);
 
-      try {
-        return this.nodesRepository.save({
-          ...existingNode,
-          ...mapClientNodeToMmpNode(clientNode, mapId),
-          lastModified: new Date(),
-        })
-      } catch(error) {
-        this.logger.error(`${error.constructor.name} - Failed to update node ${existingNode.id}: ${error}`)
-        return Promise.reject(error)
-      }
+    try {
+      await this.nodesRepository.upsert({
```

Comment:
are you sure upsert does not return the new entity? do we really need the find call?